### ISSUE TITLE
Multiple typos in assertion syntax in (?(assertion)YesPattern|NoPattern)

### ISSUE THREAD

**Issue by Medvedishce** [notepad-plus-plus/npp-usermanual#113]
See here: [Control flow](https://npp-user-manual.org/docs/searching/#control-flow)

Inside of _assertion_ syntax part of 
- `(?(assertion)YesPattern|NoPattern)` there are multiple typos.
Here are how it is (bold - typos):
  - `(ℕ)` ⇒ true if ℕth unnamed group was previously defined

  - `(`**`?`**`<name>)` or `(`**`?`**`'name')` ⇒ true if group called name was previously defined

  - `(?=lookahead)` ⇒ true if the lookahead expression matches

  - `(?!lookahead)` ⇒ true if the lookahead expression does not match

  - `(?<=lookbehind)` ⇒ true if the lookbehind expression matches

  - `(?<!lookbehind)` ⇒ true if the lookbehind expression does not match

  - **`(?`**`(R)`**`)`** ⇒ true if inside a recursion

  - **`(?`**`(Rℕ)` ⇒ true if in a recursion to subexpression numbered ℕ

  - **`(?`**`(R&name)`**`)`** ⇒ true if in a recursion to named subexpression name

- and how it must be:
[boost_regex.syntax.perl_syntax.conditional_expressions](https://www.boost.org/doc/libs/1_70_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html#boost_regex.syntax.perl_syntax.conditional_expressions)
  - `(ℕ)` ⇒ true if ℕth unnamed group was previously defined

  - `(<name>)` or `('name')` ⇒ true if group called name was previously defined

  - `(?=lookahead)` ⇒ true if the lookahead expression matches

  - `(?!lookahead)` ⇒ true if the lookahead expression does not match

  - `(?<=lookbehind)` ⇒ true if the lookbehind expression matches

  - `(?<!lookbehind)` ⇒ true if the lookbehind expression does not match

  - `(R)` ⇒ true if inside a recursion

  - `(Rℕ)` ⇒ true if in a recursion to subexpression numbered ℕ

  - `(R&name)` ⇒ true if in a recursion to named subexpression name

P.S: 
All this typos were found when I painfully did example for #112 so all corrections tested before.
P.P.S.:
In the [boost_regex.syntax.perl_syntax.conditional_expressions](https://www.boost.org/doc/libs/1_70_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html#boost_regex.syntax.perl_syntax.conditional_expressions) lookbehind assertions not defined, but I tested them and all working.

**Comment by pryrt** [comment]
Thanks for your inputs.  I am working on fixes to this and #112.  

But I have a question for you: As a user looking up the conditional-expression assertions, would it be easier to understand as it currently is (with your suggested fixes for the typos)?  Or would it be easier to understand with something like
> * `(?(ℕ)YesPattern|NoPattern)` ⇒ true if ℕth unnamed group was previously defined
> * `(?(<name>)YesPattern|NoPattern)` or `(?('name')YesPattern|NoPattern)` ⇒  true if group called name was previously defined
> * ...

Does repeating the whole conditional every time make it more readable? Or does that just look like it's repeating itself unnecessarily?

**Comment by Medvedishce** [comment]
When I started to think about corrections, I thought about two approaches:
1) To leave introduction as is and change all assertions descriptions to `(`_`assertion`_`)` style.
2) To slightly change introduction and and change all assertions descriptions to `(?(`_`assertion`_`)...` style.
At the and I've selected first one.

I think that existing preamble introduction before the assertions list have no issue. Look at me - I was ABSOLUTELY certain that I must use `(?(?'LEQ250')in a box|in a barrel)` construct exactly as it was described. So I think that problem was not in the preamble part but due to explosive mix of two different styles with addition of couple of typos in description part.  

P.S.: 
In case you will decide to use extended syntax to eliminate ambiguous preamble I will suggest to use slightly shorter `(?(`_`assertion`_`)...` style. We can see application of full extended syntax on the [boost_PCRE](https://www.boost.org/doc/libs/1_70_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html#boost_regex.syntax.perl_syntax.conditional_expressions) page and it's really heavyweight, hard to read bunch of text.
